pkg/test/framework/resource/context.go

```diff
@@ -20,20 +20,37 @@ import (
 	"istio.io/istio/pkg/test/util/yml"
 )
 
-type ConfigOptions struct {
-	NoCleanup bool
-	Wait      bool
+type ApplyOptions struct {
+	Cleanup CleanupStrategy
+	Wait    bool
 }
 
-type ConfigOption func(o *ConfigOptions)
+type ApplyOption func(o *ApplyOptions)
```

Comment:
LGTM, just one question: why not use struct directly,

Reply:
I could go either way, really. I think my original idea was that most of the time when applying config, you don't need to specify anything, so the call would just be `cfg.Apply()`. If it were a struct, the call site would look like `cfg.Apply(resource.ApplyOptions{})`. TBH I don't feel strongly either way.

Reply:
That is many api does  , even client-go